feat(SessionSerializer): Transitional serializer write to JSON object

maxiuyuan opened the pull request:
second part of https://github.com/getsentry/sentry/pull/31144

Transitional Serializer now writes to JSON object

one major change is that datetime objects are now changed to its utc_timestamp. Looking through the code base at all instances of `request.sessions`, it doesn't seem the date object is used

#sync-getsentry

maxiuyuan committed:
transitional serializer writes to JSON object

maxiuyuan committed:
Merge branch 'transitional-serializer' into transitional-serializer-write

maxiuyuan committed:
added datetime to be serialized

maxiuyuan committed:
use utc_timestamp

JoshFerge commented on src/sentry/utils/transitional_serializer.py:
```diff
@@ -11,9 +12,11 @@ def __init__(self) -> None:
-        # will have the write as pickle obj for now, will change to json in seperate deploy
-        # return self.json_serializer.dumps(obj)
-        return self.pickle_serializer.dumps(obj)
+        for key in obj:
```
what if the timestamp is in a nested part of the object?

JoshFerge replied:
how will this work when loading? don't we need changes for that?

maxiuyuan replied:
good point on the nested part, ill come up with a way on how to do that. using the same logic, i can implement a way to load it back to a datetime obj using the same logic in the loads fn

JoshFerge replied:
okay. an alternative would be to convert the callers to pass in a raw timestamp to avoid having to convert it here -- i'll let you decide.

maxiuyuan replied:
the way i thought of dealing with a nested datetime obj is to recursively call a helper fn if a type dict is in the value.  then once it loads it calls another helper fn to go thru with the same logic of converting datetime to timestamp but the other way around to turn it back. i've pushed up the logic, lmk if its good or a more efficient method should be used

maxiuyuan committed:
look for nested datetimes and turn timestamp back to datetime obj on load

maxiuyuan committed:
wrong helper fn call

maxiuyuan committed:
added metrics

ceorourke committed:
ref(sentry apps): Better error message for text component defaults (#31258)

* ref(sentry apps): Better error message for text component defaults

maxiuyuan committed:
Merge branch 'transitional-serializer' into transitional-serializer-write

maxiuyuan committed:
rebuild pages

scttcper committed:
fix(ui): Add seconds to DateTime for users w/ 24 hour clock (#31266)

JonasBa committed:
test(profiling): add position indicator renderer tests (#31217)

JonasBa committed:
feat(profiling): add selectedFrameRenderer (#31176)

JonasBa committed:
feat(profiling): add text renderer (#31108)

* feat(textrenderer)): add class and binary search util fn

* feat(string): add utility fn to trim string

text renderer wip

* test(profiling): add textRenderer tests

* ref(profiling): update collision code and code review comments

ahmedetefy committed:
fix(metrics-perf-table): Handles error when table is semi-populated (#31251)

matejminar committed:
ref(button): Consolidate label and aria-label (#31215)

jjbayer committed:
feat(metrics): Tag sessions_v2 query with relative end time (#31247)

This will allow us to see if differences between the sessions and
metrics backend correlate with the end time of the query.

jan-auer committed:
feat(outcomes): Dispatch billing outcomes to a dedicated topic (#31175)

Outcomes are used for stats, debugging, quota enforcement, and spike protection.
The last two categories run exclusively in Sentry SaaS and require timely
consistent handling.

To ensure this, this PR adds the possibility to send ACCEPTED and RATE_LIMITED
outcomes to a dedicated topic. Since this is optional, this will not be used in
development and self-hosted Sentry.

jjbayer committed:
fix(releasehealth): Shorter tag names in duplex (#31274)

Shorten tag keys to <= 32 characters.

narsaynorath committed:
ref(dashboard-layout): Create layout utils file (#31240)

Collect the dashboard layout helpers into one file to reduce code clutter

jjbayer committed:
fix(sessions): Order results by timestamp and log error if snuba limit exceeded (#31214)

As described in https://getsentry.atlassian.net/browse/ISSUE-1372, gaps
occur in sessions_v2 time series when the number of releases is large.
This seems to be caused by the fact that snuba applies a default limit
of 1000. The sessions API queries these series without an orderBy
constraint, so a random subset of entries default to zero.

This PR logs an error if this limit is actually reached. Furthermore, we
add an order by clause to the snuba query, such that at least the most
recent part of the time series is complete.

edwardgou-sentry committed:
fix(discover): Add to Dashboard from discover use the updated select option component to match new style (#31261)

Fixes selector to use newer option style

edwardgou-sentry committed:
feat(dashboards): Force Issue field on Issue widgets and remove redundant permalink field (#31173)

Forces the Issue field to always be displayed on Issue widgets. This is because Issue table results are always grouped by the Issue # regardless of fields selected so we should always display the Issue #.
If there is only one Issue column, it is disabled/unchangeable and undeletable.

Permalink is no longer selectable in the field dropdown because it is functionally the same as the Issue column, therefore redundant.

edwardgou-sentry committed:
feat(dashboards): Widget Library Issue Widget updates (#31250)

Updates the prebuilt Issue Widget to include some additional filters to be more relevant to the user

evanh committed:
fix(snql) Update to the latest snuba SDK (#31121)

The latest version has support for higher order functions, but also had some
breaking changes that needed to be fixed.

wmak committed:
feat(dashboards): When a dashboard template is added show a toast (#31263)

- This is so its clearer that it was successful and to give more
  positive feedback when it happens

idiomaticrefactoring committed:
Update paginator.py (#31248)

refactoring code with Dictionary Comprehension which is more pythonic, concise and efficient

wedamija committed:
fix(insights): Filter out groups with a last seen older than 90 days from unresolved issue age and old groups endpoints (#31268)

Incinerator is currently broken, which means old groups are hanging around forever and appearing in
these reports. To filter them out we remove any groups with a last seen older than 90 days ago.

wedamija committed:
feat(ownership): Add feature to allow some orgs to have a higher max project ownership limit (#31231)

We have this limit in place to protect our infrastructure against problems caused by a large number
of ownership rules.

In some cases, we might want to relax this for specific orgs. This adds in a feature flag to allow
for a higher limit.

Longer term we should figure out how to make processing large number of rules more efficient. We can
probably do prefix matching for a lot of these rules, so that we don't have to scan every rule in
the tree for every path in the stack trace.

scttcper committed:
build(ui): Upgrade webpack, fix react hot reload (#31271)

wmak committed:
feat(dashboards): Fill out the remaining dashboard templates (#31264)

* feat(dashboards): Fill out the remaining dashboard templates

- This updates the Default, Backend, Frontend templates with actual
  widgets
- This changes the default template on the frontend, but hasn't updated
  the backend version of it, will update that in a separate PR

leeandher committed:
chore(codeowners): Restrict the configuration pages to non-members (#31285)

See API-2386

This PR will restrict the entire configuration pages to non-members only. Prior to this PR if you happened to know the configuration ID of your integration you could access its page even if your permissions wouldn't let you do so in the UI.

Now, trying to go to /settings/{org}/integrations/github/{configId}/ will redirect members to the last page they had access to, /settings/{org}/integrations/github/.

wedamija committed:
fix(api): Fix OrganizationTagKeyValuesEndpoint to support passing environment (#31290)

We pass in environment name to `get_tag_value_paginator_for_projects`, but we actually require an id
here. When passing an environment name we actually error here when we attempt to translate the
environment filter param.

This fix just changes the code to pass environment ids rather than names.

Commit:
feat(alert-rule-action): Optional description for alert-rule-settings (#31289)

Objective:
Allow SentryApp to provide a description above the form for Alert Rule Action. This will be optional. The description should be above the divider of the form and below the title. Will limit the description to 140 characters.

wedamija committed:
logs(weekly_reports): Add extra log to `prepare_reports` (#31301)

Investigating this issue again. Seeing weird things in the logs, and want to confirm that this task
finishes, and how long it takes.

maxiuyuan committed:
fixed /stats_v2/ endpoint returns zero for a lot of time slices (#31296)

problem: https://getsentry.atlassian.net/browse/ER-913

The reason why this was happening was in make_timeseries, how the rows
was sorted was changed. Tbjs caused the row[ts_col] == ts to not match
in most cases. In turn this set row to None which resulted in a 0 in the
series field for the date that it was compared to.

The fix is to revert it back to how the rows were being sorted before.

beezz committed:
Make sure to stop workers on main process crash (#31249)

* Make sure to stop workers on main process crash

* Avoid skip_models setup in each task iteration

narsaynorath committed:
ref(dashboard-layout): Use widgets as source of truth for layout (#31279)

Reduce the amount of indirection when working with layouts. We can read
the layout from the widgets and propagate changes back to the modifiedDashboard
when a layout change has occurred

matejminar committed:
feat(ui): Add includeSessions to crash alerts tag values (#31277)

0Calories committed:
fix(ui): Prevent suspect span table from displaying frequencies above 100% (#31293)

* Add formatting adjustments

* Adjust commented explanation in suspectSpansTable

Co-authored-by: Tony Xiao <txiao@sentry.io>

scttcper committed:
chore(ui): Add list + list item stories (#31298)

edwardgou-sentry committed:
fix(dashboards): Fixed incorrect widget card for Top N widgets when creating from Discover (#31260)

fixed issues with widget card in widget modal not querying proper fields when attempting to add top n widget from discover. Top N discover queries need to pass their selected columns and yaxis in the Widget's field attribute due to how Top N widgets are implemented.

fixed issue with widget fields sometimes entering broken states when switching display types from Discover. Refactored code area around Widget Modal from discover.

Commit:
feat(notifications): overage slack notifications backend (#31283)

This PR adds backend support for the quota category and quota sub-categories (quotaErrors, quotaTransactions,quotaAttachments,quotaWarnings). The idea is that on one page a user could modify all the quota related settings together rather than having multiple different pages. Note that a front-end PR will have to be added to start using the fields. These new fields will enable us move quota notifications to the notification platform so we can add Slack messages.

wedamija committed:
perf(weekly_reports): Only fetch org id when getting ready to prepare reports (#31304)

We don't need to bring the whole org into memory here. To make this function run faster, just fetch
the id and use that directly.

nikhars committed:
fix(consumer-lag): Remove header optimization for some cases (#31292)

When the header `transaction-forwarder` is present in the kafka headers,
snuba errors and transaction consumers apply an optimization where
they ignore messages which are not destined for them. But the way
they ignore it cause offsets not to be committed often if there are
only 1 type of messages on the shared topic. Removing the header if the
option "eventstream:kafka-headers" is not set. This would resolve
https://github.com/getsentry/self-hosted/issues/1204

Commit:
adds utility function for setting up Slack (#31314)

scttcper committed:
fix(workflow): Use sidebar section in issues tag filter (#31229)

scttcper committed:
feat(workflow): Show disabled create ownership rule button (#31227)

maxiuyuan committed:
no longer needs to serialize datetime obj

maxiuyuan commented:
no longer needs to serialize datetime obj after https://github.com/getsentry/sentry/pull/31322

mgaeta committed:
fix(notifications): Fix `get_user_subscriptions_for_groups()` (#31223)

vuluongj20 committed:
ref(ui): Get button styles from theme object (#31204)

* ref(ui): Get button styles from theme object

 - Add a `form` property to the theme object containing common styles for form inputs and buttons.
 - Update `button.tsx` to use the newly-added `form` prop from `theme`

evanpurkhiser committed:
ref(pageFilters): More shared-type organization (#31267)

Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com>

evanpurkhiser committed:
ref(js): Use DateString in getPeriod (#31331)

davidenwang committed:
fix(gsh): Allow absolute date when navigating (#31299)

Recent changes to GSH may have introduced a bug where modifying the GSH's date through interacting with discover graphs does not seem to take effect, or only take effect when performed multiple times.

ceorourke committed:
ref(pagerduty): Add logging for resolving metric alerts (#31294)

* ref(pagerduty): Add logging for resolving metric alerts

evanpurkhiser committed:
Revert "ref(pageFilters): More shared-type organization (#31267)" (#31333)

vuluongj20 committed:
ref(ui): Use light text for release projects panel's header (#31324)

wedamija committed:
fix(tests): Make `TeamIssueBreakdownTest` stable (#31334)

This broke randomly on master, likely just fails at specific times.

wedamija committed:
debug(weekly_reports): Add extra log to `prepare_reports` (#31336)

I never saw `finish_prepare_report` in the logs this week. Want more detail about how far we get
through - this will add around 100 logs per run so not huge.

ahmedetefy committed:
feat(release): Adds btree index on `Release` version (#31243)

* feat(release): Adds btree index on Release version

Adds a btree index on field `version` of model
`Release` to be able to perform prefix search on
release names

* Rename index name

matejminar committed:
feat(ui): Add sorting and pagination to performance metrics table (#31275)

untitaker committed:
fix(release_health): Fix another instance of wrong granularity (#31306)

shruthilayaj committed:
ref(discover): Dataset-specific config to specify fields and filters (#31272)


Introduce a dataset-specific configuration that generates
the available fields, functions and filters for the dataset
so that the logic doesn't live inside QueryBuilder. Furthermore,
remove the multiple levels of inheritance and move the
related functions inside QueryBuilder since a lot of
complexities are abstracted away by the dataset config.
Note, not removing the QueryFilter, QueryFields and QueryBase
class yet because they are inherited or called elsewhere.
Will be removed as a follow up.

matejminar committed:
fix(ui): Fix .NET stacktrace path (#31346)

edwardgou-sentry committed:
feat(dashboards): Issue widget count tooltips clickable and redirect to discover (#31297)

Makes the count tooltip for issue widgets clickable and redirects the user to discover

edwardgou-sentry committed:
feat(dashboards): Increase table widget rows (#31328)

Adds a tableItemLimit prop to allow configuring row limit
Increases the table row limit for dashboard displayed table widgets to 20. Preview widget cards remain at a 5 row limit.

matejminar committed:
fix(ui): Change the order of release comparison toggle (#31350)

priscilawebdev committed:
fix(failure-rate-metrics-in-peformance): Fix wrong manually calculation (#31347)

udameli committed:
feat(trends): Implicitly set trend parameter as filter value in query (#31112)

* feat(trends): Implicitly set trend parameters as filter values in the query

For the performance trends view, check if current parameter is a web vital parameter. If so, add its value as an additional condition filter to the query

Fixes VIS-1273

* test(trends): Test web vital parameters being added as additional condition filters to query

JoshFerge committed:
fix(auth): remove saml expiry logic (#31322)

scttcper committed:
fix(teamStats): Re-arrange unresolved, adjust titles (#31291)

ceorourke committed:
ref(webhooks): Don't raise errors for 301s (#31327)

* ref(webhooks): Don't raise errors for 3XX

mikellykels committed:
fix(ui): Handle project selection when logger clicked  (#31307)

* fix(ui): Handle project selection when logger clicked

Fix issue where when you click the logger from the Issue Stream, it would go to the Custom Search tab but remove the project from the global header. If you don't have the multiple project feature, it would display an error. This fix should persist the selected project.

FIXES ISSUE-1381

* update for multiple projects

* refactor to use globalselection link

wmak committed:
feat(discover): Label breakdown & measurements as fields (#31355)

- This is cause they're effectively fields, even though they're more
  specific fields and to try to avoid additional confusion
- This also reorders the dropdowns so measurements & breakdowns appear
  before tags, rather than after them

evanpurkhiser committed:
ref(pageFilters): More shared-type organization (#31338)

evanpurkhiser committed:
ref(js): Move DATA_CATEGORY_NAMES to constants (#31339)

maxiuyuan committed:
Merge branch 'master' into transitional-serializer-write
